File: typescript/packages/app/src/seed.ts
/** Deterministic seed reporting. */

import { METRICS } from "@atelier/bench";

import type { AppState } from "./state";

/** One-line description of the frozen dataset, printed by `seed`. */
export function summarize(state: AppState): string {
    return [
        `seeded: ${state.data.customers.length} customer(s)`,
        `${state.data.orders.length} order(s)`,
        `${state.data.parts.length} part(s)`,
        `revenue ${state.data.revenueCents()}c`,
    ].join(", ");
}

/** Every registered metric against the seed, in registry order. */
export function metricLines(state: AppState): string[] {
    return METRICS.map((entry) => `${entry.key} = ${entry.formatted(state.data)}`);
}

File: rust/crates/atelier-app/src/seed.rs
//! Deterministic seed reporting.

use crate::state::AppState;

/// One-line description of the frozen dataset, printed by `seed`.
pub fn summarize(state: &AppState) -> String {
    format!(
        "seeded: {} customer(s), {} order(s), {} part(s), revenue {}c",
        state.data.customers.len(),
        state.data.orders.len(),
        state.data.parts.len(),
        state.data.revenue_cents(),
    )
}

/// Every registered metric against the seed, in registry order.
pub fn metric_lines(state: &AppState) -> Vec<String> {
    atelier_bench::METRICS
        .iter()
        .map(|metric| format!("{} = {}", metric.key(), metric.formatted(&state.data)))
        .collect()
}
